File: backend/LeaveManagerAPI/Models/LeaveBalance.cs
namespace LeaveManagerAPI.Models
{
    public enum LeaveType
    {
        AnnualLeave,
        SickLeave,
        Unpaid
    }
    public class LeaveBalance
    {
        public int Id { get; set; }

        public string UserId { get; set; } = string.Empty;

        public User User { get; set; }

        public int Year { get; set; }

        public LeaveType Type { get; set; }

        public int TotalDays { get; set; }

        public int UsedDays { get; set; }

        public int RemainingDays => TotalDays - UsedDays;
    }
}

File: frontend/src/app/models/leave-balance.ts
export interface LeaveBalanceDto {
    type: string;
    totalDays: number;
    usedDays: number;
    remainingDays: number;
}

export interface LeaveBalanceResponse{
    id: number,
    userId: string,
    userFullname: string,
    year: number,
    type: string,
    totalDays: number,
    usedDays: number,
    remainingDays: number
}

export interface CreateLeaveBalanceRequest{
    userId: string,
    year: number,
    type: string,
    totalDays: number,
    usedDays: number
}

export interface UpdateLeaveBalanceRequest {
    totalDays: number
}

export interface FilterLeaveBalance {
    userFullname?: string | null,
    year?: number | null,
    type?: string | null
}